xmind/tests/test_element.py

```diff
@@ -0,0 +1,347 @@
+        self.assertEqual(_element.getAttribute('t1'), None)
+        _node.getAttribute.assert_not_called()
+        _node.hasAttribute.assert_called_once_with('t1')
+        _get_local_name.assert_called_once_with(_element, 't1')
+
+    def test_getAttribute_node_hasNoAttribute_localName_differs(self):
+        """Tests getAttribute method when node has no attribute and localName differs from attribute name"""
+        _get_local_name = self._init_patch_with_name(
+            "_get_local_name", "xmind.core.Element.getLocalName", return_value='t2', autospec=True)
+        _node = MagicMock()
+        _node.getAttribute.return_value = 12
+        _node.hasAttribute.side_effect = [False, True]
+        _element = Element(_node)
+        with patch.object(Element, 'getAttribute', wraps=_element.getAttribute) as _mock:
+            self.assertEqual(_element.getAttribute('t1'), 12)
+
+        self.assertEqual(_mock.call_count, 2)
+        self.assertListEqual(_mock.call_args_list, [call('t1'), call('t2')])
+        _node.getAttribute.assert_called_once_with('t2')
+        self.assertEqual(_node.hasAttribute.call_count, 2)
+        self.assertListEqual(_node.hasAttribute.call_args_list, [
+                             call('t1'), call('t2')])
+        _get_local_name.assert_called_once_with(_element, 't1')
+
+    def test_setAttribute_attr_value_None(self):
+        """Tests setAttribute when attr_value is None"""
+        _node = MagicMock()
+        _node.hasAttribute.return_value = False
+        _element = Element(_node)
+        _result = _element.setAttribute('some')
+        self.assertIsNone(_result)
+        _node.setAttribute.assert_not_called()
+        _node.removeAttribute.assert_not_called()
+        _node.hasAttribute.assert_called_once_with('some')
+
+    def test_setAttribute_attr_value_is_not_None(self):
+        """Tests setAttribute when attr_value is not None"""
+        _node = MagicMock()
+        _element = Element(_node)
+        _result = _element.setAttribute('some', '2')
+        self.assertIsNone(_result)
+        _node.setAttribute.assert_called_once_with('some', '2')
+        _node.removeAttribute.assert_not_called()
+        _node.hasAttribute.assert_not_called()
+
+    def test_setAttribute_attr_value_is_None_remove_attribute(self):
+        """Tests setAttribute when attr_value is None, removeAttribute called"""
+        _node = MagicMock()
+        _node.hasAttribute.return_value = True
+        _element = Element(_node)
+        _result = _element.setAttribute('some')
+        self.assertIsNone(_result)
+        _node.setAttribute.assert_not_called()
+        _node.removeAttribute.assert_called_once_with('some')
+        _node.hasAttribute.assert_called_once_with('some')
+
+    def test_createElement(self):
+        """Tests createElement method, returns nothing"""
+        _node = MagicMock()
+        _element = Element(_node)
+        _result = _element.createElement('some')
+        self.assertIsNone(_result)
```

Comment:
I see that createElement method is empty. it has pass inside. why should we write test for it? and if we are writing, then should we test it without parameters etc?

Reply:
check test_excessive_parameters test